meetings/2018-10-17.md

```diff
@@ -0,0 +1,166 @@
+    * Affects our code, as code that comes in uses non-const references for example using 
+      Containers.
+
+  * Anna
+    * We pretty much follow the style that V8 uses even if it is not in the style guides
+    * Understands where Rafael is coming from and making it easy for newcomers, there is still
+      value in making it easy for reviewers as this makes it obvious when something is changed
+      or not.
+    * PR is capturing what we have been doing, but which has not yet been formalized.
+
+  * Nikita, how would accepting this affect our code?
+  
+  * Gabriel, does it make sense to add additional reasons in the PR itself so that its clearer why
+    the guideline is there. 
+
+  * Rafael
+    * in favor of documenting as he was looking for rational/reasoning to be explained
+    * recently we landed general guidelines which say follow core guidelines unless they 
+      conflict with google guidelines or our own. 
+    * still missing motivation to be so cut and dry about pointers being a better choice
+
+  * Nikita
+    * Do we have any code that does not follow that rule, 
+
+  * Rafael
+    * yes there are many v8 APIs that required you use non-const references
+    * we have been fairly lax in the past of enforcing style guide as it has not been well formalized
+      in the past
+ 
+  * Joyee
+    * In some ways our C++ is basically V8 C++, because it has been based on using
+      the V8 APIs
+    * Style should be aligned with V8 style because of this dependency.
+   * Most of non-const references are overloads, most of the time they still use pointers
+   * We could probably list out specific examples as opposed to saying pointers are always
+     Preferred.
+  * Rich kick back to github
+
+  * Nikita
+    * If we relax the rule (based on it being defacto), what can we change in our existing
+      code base that would improve it?
+
+  * Rafael
+    * use of for in ?over container as opposed to having to doing bounds checking.
+    * for each in newer compilers
+    * avoids needs for null checks in some cases
+    * Don’t believe it will be very different from V8, but don’t agree completely we should 
+      infer what we think the v8 team was trying to get at.
+
+  * Anna 
+    * One of main arguments, non const refs indicate that API is going to change something,
+      but the same is true for pointers as well.
+  
+  * Ali propose
+    * Document what we have now (existing practice)
+    * Have a new issue/PR to change it if we want to agree to do something different.
+
+  * Take it back to GitHub.
+
+  * Rafael, high barrier to entry for our C++ code, small number of people maintaining.  Would
+    like to have larger group participating.
+
+* fs: behaviour of readFile and writeFile with file descriptors [#23433](https://github.com/nodejs/node/issues/23433)
+  * please review, added by thefourtheye for which time is bad, so we’ll discuss next week
```

Comment:
Sorry about this. I tried hard but couldn't stay awake.

Reply:
we can't expect people to stay up in the middle of the night!